hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainerCheck.java

```diff
@@ -263,10 +265,25 @@ private ScanResult scanData(DataTransferThrottler throttler,
               if (getBlockDataFromDBWithLock(db, block) != null) {
                 // Block was not deleted, the failure is legitimate.
                 return result;
-              } else if (LOG.isDebugEnabled()) {
+              } else {
+                // If schema V3 and container details not in DB or
+                // if containerDBPath is removed
+                if ((onDiskContainerData.hasSchema(OzoneConsts.SCHEMA_V3) &&
```

Comment:
Would it be simpler to check if the container entry is removed from the container map in memory? What order does a "delete container" work? For Schema V3, it needs to at least:

1. Remove the entries from the DB
2. Remove the files from disk
3. Remove the entry from the in memory map.

I also wonder about locking - is there a "container lock"? Does the scanner use it while scanning, and the delete flow use it while deleting?

The solution here seems OK, but these are the things i thought of as I looked at this.

Reply:
Thanks for the review @sodonnel You're right. This is the container delete is handled in `KeyValueHandler.java`

```
1. containerSet.removeContainer(containerId); // removes container from mem map
2. KeyValueContainerUtil.removeContainerDB; // removes container entries from DB
3. KeyValueContainerUtil.moveToDeletedContainerDir; // removes container dir from disk.
```
The scanner does attempt to read the container with a lock if the scanBlock fails once on L265. We try to check if the container is deleted if this read with lock fails. Is this what you meant? It would be easier to check the container state is removed from the mem map instead of checking the DB. I will update the PR.

Reply:
However, It doesn't seem like there is an easy way to read the containerMap in DN memory from the scrubber.

Reply:
Conflicts resolved.